ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs

```diff
@@ -90,10 +94,11 @@ instance Crypto c => HasTxs (ShelleyBlock c) where
 instance Crypto c => ToCBOR (GenTx (ShelleyBlock c)) where
   -- No need to encode the 'TxId', it's just a hash of the 'SL.TxBody' inside
   -- 'SL.Tx', so it can be recomputed.
-  toCBOR (ShelleyTx _txid tx) = toCBOR tx
+  toCBOR (ShelleyTx _txid tx) = wrapCBORinCBOR toCBOR tx
 
 instance Crypto c => FromCBOR (GenTx (ShelleyBlock c)) where
-  fromCBOR = mkShelleyTx <$> fromCBOR
+  fromCBOR = fmap mkShelleyTx $ unwrapCBORinCBOR
+    $ (. Full) . runAnnotator <$> fromCBOR
 
```

Comment:
This was the extent of the serialization changes @mrBliss

Reply:
Great :+1:. Can you make a separate PR for this that we can merge after #1947 and #1928?